pkg/pillar/containerd/containerd.go

```diff
@@ -229,22 +178,48 @@ func CtrPrepare(containerPath string, ociFilename string) error {
 	log.Infof("mountPoints %+v execpath %+v cmd %+v workdir %+v env %+v",
 		mountpoints, execpath, cmd, workdir, unProcessedEnv)
 
-	if err != nil {
-		return fmt.Errorf("ctrPrepare: unable to get image: %v config: %v", ctrdImage.Name(), err)
-	}
-	// containerd.NewImageWithPlatform(client, i, platforms.Only(platform))
+	// unpack the rootfs Image if needed
 	unpacked, err := ctrdImage.IsUnpacked(ctrdCtx, defaultSnapshotter)
 	if err != nil {
-		return fmt.Errorf("ctrPrepare: unable to get image metadata: %v config: %v", ctrdImage.Name(), err)
+		return fmt.Errorf("snapshotPrepare: unable to get image metadata: %v config: %v", ctrdImage.Name(), err)
 	}
 	if !unpacked {
 		if err := ctrdImage.Unpack(ctrdCtx, defaultSnapshotter); err != nil {
-			return fmt.Errorf("ctrPrepare: unable to unpack image: %v config: %v", ctrdImage.Name(), err)
+			return fmt.Errorf("snapshotPrepare: unable to unpack image: %v config: %v", ctrdImage.Name(), err)
 		}
 	}
-	if err = ctrCreate(filepath.Base(containerPath), ctrdImage); err != nil {
-		return fmt.Errorf("ctrPrepare: failed to create container %s, error: %v", containerPath, err.Error())
+
+	// use rootfs unpacked image to create a writable snapshot with default snapshotter
+	diffIDs, err := ctrdImage.RootFS(ctrdCtx)
+	if err != nil {
+		log.Errorf("Could not load rootfs of image: %v. %v", ctrdImage.Name(), err)
+		return fmt.Errorf("snapshotPrepare: Could not load rootfs of image: %v. %v", ctrdImage.Name(), err)
+	}
+
+	snapshotter := CtrdClient.SnapshotService(defaultSnapshotter)
```

Comment:
Replace this with `containerd.DefaultSnapshotter` as above.

Reply:
Same question for you as above @deitch